### PR TITLE
Editor: Make "open 2d/3d/script editor" shortcuts configurable.

### PR DESCRIPTION
**Pull request by 27thLiz**
Also adds a shortcut for opening the AssetLib (defaulting to F4).

### CODE REVIEW

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -6111,6 +6097,11 @@ EditorNode::EditorNode() {
 	_dim_timer->set_wait_time(0.01666f);
 	_dim_timer->connect("timeout", this, "_dim_timeout");
 	add_child(_dim_timer);
+
+	ED_SHORTCUT("editor/editor_2d", TTR("Open 2D Editor"), KEY_F1);
```
Comment (akien-mga) [inline editor/editor_node.cpp]: Please also take the opportunity to put the shortcuts back to F2 for 2D and F3 for 3D, the current ones always bothered me :P
F1 should likely open the quick search dialog of the help.

**Comment by neikeq** [comment]
I would love to see is two shortcuts (even if they are set to nothing by default), to switch to the next and previous editor. Similar to how you switch between workspaces in some desktop environments. e.g.:

<kbd>Ctrl</kbd>+<kbd>Alt</kbd>+ <kbd>Left</kbd> / <kbd>Right</kbd>: :arrow_right_hook: 2D <-> 3D <-> Script <-> AssetLib :leftwards_arrow_right_hook:

**Comment by bojidar-bg** [comment]
Uhm. <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+ <kbd>Left</kbd> / <kbd>Right</kbd> are already used by sane linux distros for "Switch workspace left / right"

**Comment by groud** [comment]
Hello,
How about _Ctrl+Tab_ and _Ctrl+Shift+Tab_, as Firefox does to switch between tabs ?

**Comment by Hinsbart** [comment]
Ah, `ctrl + tab` and `ctrl+shift+tab` are already taken for switching scene tabs.
Any alternatives, or should I add the shortcuts and leave them unassigned?

**Comment by HummusSamurai** [comment]
@Hinsbart I think that's definitely a good idea for now, since we (the people who want this feature pretty bad) can start using it while the collaborators decide on sane defaults for the keybinding.

**Comment by Hinsbart** [comment]
Ok, will leave them unassigned for now. The `tab` key currently doesn't work with shortcuts anyway, will open an issue about this..

**Comment by Hinsbart** [comment]
Alright, updated.

File: editor/editor_node.cpp [hunk 2]
```diff
@@ -6111,6 +6128,14 @@ EditorNode::EditorNode() {
 	_dim_timer->set_wait_time(0.01666f);
 	_dim_timer->connect("timeout", this, "_dim_timeout");
 	add_child(_dim_timer);
+
+	ED_SHORTCUT("editor/editor_2d", TTR("Open 2D Editor"), KEY_F2);
+	ED_SHORTCUT("editor/editor_3d", TTR("Open 3D Editor"), KEY_F3);
+	ED_SHORTCUT("editor/editor_script", TTR("Open Script Editor"), KEY_F4);
+	ED_SHORTCUT("editor/editor_help", TTR("Search Help"), KEY_F1);
+	ED_SHORTCUT("editor/editor_assetlib", TTR("Open Asset Library"), KEY_F5);
```
Comment (akien-mga) [inline editor/editor_node.cpp]: Will conflict with `Run`, no? On IRC we mentioned it should be kept unassigned.

Comment (Hinsbart) [inline editor/editor_node.cpp]: Oops, guess I forgot to amend this change..

Comment (Hinsbart) [inline editor/editor_node.cpp]: ok

**Pull request merged**